manifests/job.pp

```diff
@@ -43,6 +44,9 @@
       jobname => $jobname,
     }
   } else {
+    if $config == undef and $source == undef and $template == undef {
+      fail('You should at least specify one of the $config, $source or $template param')
+    }
```

Comment:
In the RSpec (`spec/defines/jenkins_job_spec.rb`) many of the examples specify parameters such as:

 ```ruby
let(:params) {{ :enabled => 0 , :config => '' }}
```

This is basically passing an empty string through as the `$config` parameter, I suppose Puppet is treating an empty string as an `undef`.